spec/predicates/release.md

```diff
@@ -0,0 +1,110 @@
+
+If these release attestations are optionally published to a transparency log, package authors (or other interested parties) can monitor when a new version of a package is released.
+
+The release attestation provides integrity for the release artifacts, but it does not provide availability as it does not require the registry to serve an attested release artifact.
+
+These use cases are not hypothetical; both of them are the case today for npm's [build provenance feature], which includes a [publish attestation]. Note that while npm calls this a publish attestation, calling it a release attestation better reflects that it's coming from the package registry. Publishing often refers to an author sending content to the registry, as in [PyPI's trusted publishers feature].
+
+## Prerequisites
+
+This predicate depends on the [in-toto attestation framework], as well as the [purl-spec] for identifying packaging ecosystems.
+
+Perhaps surprisingly, this predicate does not depend on [SLSA Provenance], but they are better together. In an ecosystem where some (but not all) packages have SLSA Provenance and you query by artifact hash, you can't tell the difference between an artifact that has been tampered with and one that does not yet have SLSA provenance. If the ecosystem has release attestations, you can give authoritative answers to what artifacts make up a given release version, and what hashes those artifacts should have.
+
+## Model
+
+This predicate is for the final stages of the software supply chain, where consumers are looking for attestations that the software has not been tampered with during distribution.
+
+If a previously attested release is later determined to be untrustworthy, a registry MAY stop serving the artifacts associated with that release.
+
+If a registry supports immutable releases, there SHOULD be one release attestation for a given `predicate.purl`.
+
+Even if the registry does not support immutable releases, the attestation subject SHOULD include all the artifacts associated with the release at that time; otherwise it will be unclear if an artifact was later removed from a release. For example, if a release version has an initial release attestation with artifact A, and then later has a release attestation with only artifact B, that SHOULD be interpreted as the release version now only containing artifact B.
+
+## Schema
+
+### Fields
+
+- **`subject.name`** string
+  - The filename of the artifact as it would appear on disk.
+
+- **`predicate.purl`, required** string (ResourceURI)
+  - A purl uniquely identifying a specific release name and version from a package registry.
```

Comment:
I haven't made any changes here yet, but I'm still open to it.

I'm not sure what we mean, exactly, by not resolvable. Do we mean that purls aren't understood by tools like `curl` or a web browser? Or do we mean that just because you have a purl the registry might not serve you that content (if the registry is down, if it determines the content is malicious, or for any other reason)? Or is there another meaning I'm missing?

Reply:
Good point. I was hoping to find more clarity in the spec, but it only states that the "URI SHOULD resolve to the artifact". So I take that to mean that "unresolvable" does cover both a purl that isn't understood by a web tool and/or may point to a non-existent entry in a registry. We really ought to update the spec to clarify this, though, since you're right to point out that "unresolvable" is imprecise.

Reply:
After we update the spec, we can evaluate if we need to clarify here.